src/ringbuffer_utils.cc

```diff
@@ -59,6 +75,14 @@ DDRes ring_buffer_create(size_t buffer_size_page_order,
   pevent->custom_event = custom_event;
   pevent->ring_buffer_type = ring_buffer_type;
   pevent->ring_buffer_size = buffer_size;
+
+  if (ring_buffer_type == RingBufferType::kMPSCRingBuffer) {
```

Comment:
The workflow is slightly strange as we `mmap` and `munmap` the buffer within the init.
Though as discussed we do not have the mmap within the creation workflow.

Reply:
Perhaps we can add a comment to justify the logic

Reply:
Done